Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -300,26 +292,34 @@ func lint(fileAt url: URL, target: String) throws -> LintResult {
 // 1st arg = project path
 let args = CommandLine.arguments.dropFirst()
 guard let projectPath = args.first, !projectPath.isEmpty else { print("You must provide the path to the xcodeproj as first argument."); exit(1) }
-let project = try Xcodeproj(path: projectPath)
-
-// 2nd arg (optional) = name of target to lint
-let targetsToLint: [Xcodeproj.PBXNativeTarget]
-if let targetName = args.dropFirst().first, !targetName.isEmpty {
-    targetsToLint = project.nativeTargets.filter { $0.name == targetName }
-} else {
-    targetsToLint = project.nativeTargets.filter { $0.productType == .appExtension }
-}
-
-// Lint each requested target
-var violationsFound = 0
-for target in targetsToLint {
-    let files: [Xcodeproj.PBXBuildFile] = project.buildFiles(for: target)
-    print("Linting the Build Files for \(target.name):")
-    for file in files {
-        let result = try lint(fileAt: project.resolveURL(to: file).absoluteURL, target: target.name)
-        print("  - \(project.resolveURL(to: file).relativePath) [\(result)]")
-        if case .violationsFound(let list) = result { violationsFound += list.count }
+do {
+    let project = try Xcodeproj(path: projectPath)
+    
+    // 2nd arg (optional) = name of target to lint
+    let targetsToLint: [Xcodeproj.PBXNativeTarget]
+    if let targetName = args.dropFirst().first, !targetName.isEmpty {
+        print("Selected target: \(targetName)")
+        targetsToLint = project.nativeTargets.filter { $0.name == targetName }
+    } else {
+        print("Linting all app extension targets")
+        targetsToLint = project.nativeTargets.filter { $0.knownProductType == .appExtension }
+    }
+    
+    // Lint each requested target
+    var violationsFound = 0
+    for target in targetsToLint {
+        let buildFiles: [Xcodeproj.PBXBuildFile] = project.buildFiles(for: target)
+        print("Linting the Build Files for \(target.name):")
+        for buildFile in buildFiles {
+            guard let fileURL = try project.resolveURL(to: buildFile) else { continue }
+            let result = try lint(fileAt: fileURL.absoluteURL, targetName: target.name)
+            print("  - \(fileURL.relativePath) [\(result)]")
+            if case .violationsFound(let list) = result { violationsFound += list.count }
+        }
     }
+    print("Done! \(violationsFound) violation(s) found.")
+    exit(violationsFound > 0 ? 1 : 0)
+} catch let error {
+    print("\n\nError while parsing `\(projectPath)`: \(error)")
```

Comment:
What do you think of adding the "error:" prefix, which should make Xcode report this in the navigator?

```suggestion
    print("\n\nerror: Error while parsing `\(projectPath)`: \(error)")
```

<img width="608" alt="image" src="https://user-images.githubusercontent.com/1218433/169220819-872ac94a-fc8e-427f-8b2f-7e9d8c1d1c0b.png">

Reply:
Actually we should also be able to include the path as well in the part interpreted by Xcode I think:

```suggestion
    print("\n\n\(projectPath): error: \(error)")
```

Let me try by intentionally create an error and see.

(Loved your 🥔 object, btw 😂 )

Reply:
Applied in [eb3cf3e](https://github.com/wordpress-mobile/WordPress-iOS/pull/18525/commits/eb3cf3effc1f1a1c075ea791422b90555e17336a)

PS: Note that in my experiments, Xcode does not always report the projectPath in the main "Error" UI, hence why I kept it in the error message as well to make sure it would always be visible regardless of context and Xcode being Xcode.